### PR TITLE
Dev env tooling, server package restructuring, add did:test and *.test rules

### PR DESCRIPTION
**Pull request by pfrazee**
- Adds the concepts of `did:test` and the `.test` username TLDs
- Gets the dev-env package back into a working state
- Updates to server package:
  - Adds `ServerConfig` to load and manage server configuration
  - Adds did:test and .test behaviors to createAccount
  - Adds `DidTestRegistry` which can be shared between PDS instances to facilitate lookups across servers during testing
  - Restructures the package to make importing as a library easier
  - Adds the start of a did:web route (unfinished)

### CODE REVIEW

**Comment by pfrazee** [comment]
Starting to regret putting the timestamp in the code generation output

File: docs/specs/adx/name-resolution.md [hunk 1]
```diff
@@ -29,32 +29,128 @@ async function resolveName(name: string, nameServer?: string) {
...
 
-  const res = fetch(`${origin}${WELL_KNOWN_PATH}`, {
-    headers: { Accept: 'text/plain', Host: name },
-  })
-  
-  assert(typeof res === 'string' && res.startsWith('did:'))
-  return res
-}
+```typescript
+await xrpc.service('https://alice.pds.com').todo.adx.resolveName() // => {did: 'did:plc:12345'}
```
Comment (dholms) [inline docs/specs/adx/name-resolution.md:45]: wait do you call it on `alice.pds.com` or on `pds.com`?

Comment (pfrazee) [reply docs/specs/adx/name-resolution.md:45]: There are two situations that you call `resolveName`:
- A client speaking to their own PDS (client/server API)
- Any node trying to resolve the name (names protocol)

In the client/server case, you call it against your PDS:

```typescript
await xrpc.service('https://my-pds.com').todo.adx.resolveName({name: 'alice.pds.com'})
```

In the names protocol case, you call it against the name's origin:

```
await xrpc.service('https://alice.pds.com').todo.adx.resolveName()
```

File: packages/auth/src/auth-store.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export class AuthStore implements Signer {
 
   async canSignForDid(did: string): Promise<boolean> {
     // for dev purposes
-    if (did.startsWith('did:example')) {
+    if (did.startsWith('did:test:')) {
```
Comment (dholms) [inline packages/auth/src/auth-store.ts:58]: 👍

File: packages/dev-env/src/mock/data.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+export const postTexts = [
```
Comment (dholms) [inline packages/dev-env/src/mock/data.ts:1]: ahahaha 😂

File: packages/server/src/bin.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  if (cfg.databaseLocation) {
+    db = await Database.sqlite(cfg.databaseLocation)
+  } else {
+    db = await Database.memory()
+  }
+
+  const keypair = await crypto.EcdsaKeypair.create()
+
+  const s = server(blockstore, db, keypair, cfg)
```
Comment (dholms) [inline packages/server/src/bin.ts:35]: im tempted to just stick this all in `cfg` 🤔

Comment (pfrazee) [reply packages/server/src/bin.ts:35]: I'd be fine with that

**Review by dholms — APPROVED** [review]
post-merge approval 🙇‍♂️ 🤝

**Pull request merged**